### PR TITLE
[Fix] check only db filename during merge

### PR DESCRIPTION
**Pull request by jpfeuffer**
some workflow engine might symlink or copy databases around.
Better to just check the basename during merging.

### CODE REVIEW

File: src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp [hunk 1]
```diff
@@ -32,11 +32,10 @@
 // $Authors: Julianus Pfeuffer $
 // --------------------------------------------------------------------------
 
+#include <include/OpenMS/ANALYSIS/ID/ConsensusMapMergerAlgorithm.h>
```
Comment (cbielow) [inline src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp]: `include/OpenMS` ?

Comment (jpfeuffer) [reply src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp]: hmm good question. just moved it without thinking.

File: src/tests/class_tests/openms/source/IDMergerAlgorithm_test.cpp [hunk 2]
```diff
@@ -38,6 +38,8 @@
 #include <OpenMS/FORMAT/IdXMLFile.h>
 #include <OpenMS/SYSTEM/StopWatch.h>
 #include <OpenMS/test_config.h>
+#include <QFile>
```
Comment (cbielow) [inline src/tests/class_tests/openms/source/IDMergerAlgorithm_test.cpp]: remnants from testing? Not used?

Comment (jpfeuffer) [reply src/tests/class_tests/openms/source/IDMergerAlgorithm_test.cpp]: wow quick spotting. yes, found it as well.

File: src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp [hunk 3]
```diff
@@ -377,9 +376,13 @@ namespace OpenMS
         break;
       }
       const ProteinIdentification::SearchParameters& sp = idRun.getSearchParameters();
+      String spdb = sp.db;
```
Comment (cbielow) [inline src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp]: seeing this code duplicated 4x and the code below (which looks like a tolerant version of `operator==` for ProteinIdentification::SearchParameters): can we please move this whole comparison to a member of ProteinIdentification::SearchParameters?!

**Comment by jpfeuffer** [comment]
By the way, I was kind of hoping that our File classes abstract the handling of native separators but it seems like reading a string like "C:\foo\bar.txt" on non-Windows machines is tricky.
Workarounds exist in other places as well.

File: src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp (hunk 3 above)
Comment (jpfeuffer) [reply src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp]: true. will do.

File: src/openms/source/METADATA/ProteinIdentification.cpp [hunk 4]
```diff
@@ -448,6 +492,34 @@ namespace OpenMS
         se == "ProteinInference";
   }
 
+  bool ProteinIdentification::peptideIDsMergeable(const ProteinIdentification& idRun, const String& experiment_type) const
```
Comment (cbielow) [inline src/openms/source/METADATA/ProteinIdentification.cpp]: should be `id_run`

File: src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp [hunk 5]
```diff
@@ -359,140 +357,19 @@ namespace OpenMS
...
-      {
-        //TODO actually introduce a flag for labelling modifications in the Mod datastructures?
-        //OR put a unique ID for the used mod as a UserParam to the mapList entries (consensusHeaders)
-        //TODO actually you would probably need an experimental design here, because
-        //settings have to agree exactly in a FractionGroup but can slightly differ across runs.
-        //Or just ignore labelling mods during the check
-       OPENMS_LOG_WARN << "Used modification settings from IDRun " + idRun.getIdentifier() + " does not match with the others."
-        " Although it seems to be an MS1-labeled experiment, check carefully that only non-labelling mods differ.";
-      }
+      ok = ok && ref.peptideIDsMergeable(idRun, experiment_type);
```
[the remark this answers is not in the document]
Comment (cbielow) [reply src/openms/source/ANALYSIS/ID/ConsensusMapMergerAlgorithm.cpp]: ahh.. I see now... you want to see all the warnings.
Maybe just mention that in a comment.

File: src/openms/include/OpenMS/METADATA/ProteinIdentification.h [hunk 6]
```diff
@@ -254,6 +254,7 @@ namespace OpenMS
 
       std::pair<int,int> getChargeRange() const;
       int getChargeValue_(String& charge_str) const;
```
Comment (cbielow) [inline src/openms/include/OpenMS/METADATA/ProteinIdentification.h]: side question: why does this look like a private member?

**Pull request merged**